201-machine-learning-advanced/azuredeploy.json

```diff
@@ -134,6 +166,13 @@
         "description": "Name of ApplicationInsights."
       }
     },
+    "appInsightLocation": {
+      "type": "string",
+      "defaultValue": "[if(or(equals(parameters('location'),'westcentralus'), contains(parameters('location'), 'euap')),'southcentralus',parameters('location'))]",
```

Comment:
remove the if() hear and use a location object var to map the desired location - you can remove euap from the map.

Reply:
Sorry, I didn't get it, would you please elaborate? Do you want me to create an variable and reference it here? I don't think I can reference variable in parameters section...

Reply:
See: https://github.com/Azure/azure-quickstart-templates/blob/master/azmgmt-demo/azuredeploy.json#L72